extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/handler/AstroThingHandler.java

```diff
@@ -298,22 +270,23 @@ public void triggerEvent(String channelId, String event) {
     }
 
     /**
-     * Returns the scheduler for the Astro jobs
-     */
-    public ExpressionThreadPoolExecutor getScheduler() {
-        return scheduledExecutor;
-    }
-
-    /**
      * Adds the provided {@link Job} to the queue (cannot be {@code null})
      *
      * @return {@code true} if the {@code job} is added to the queue, otherwise {@code false}
      */
-    public boolean addJobToQueue(Job job) {
-        if (job != null) {
-            return scheduledJobs.add(job);
+    public void schedule(Job job, Calendar eventAt) {
+        monitor.lock();
+        try {
+            long sleepTime = eventAt.getTimeInMillis() - new Date().getTime();
+            ScheduledFuture<?> future = scheduler.schedule(job, sleepTime, TimeUnit.MILLISECONDS);
+            scheduledFutures.add(future);
+            if (logger.isDebugEnabled()) {
```

Comment:
Shouldn't external functions called as rarely as possible while holding locks?
Should we move the logging out of the lock unlock section?